dms/models/dms_file.py

```diff
@@ -508,11 +508,10 @@ def _search(
         # operations
         if not result:
             return 0 if count else []
-        file_ids = set(result)
         directories = self._get_directories_from_database(result)
         for directory in directories - directories._filter_access("read"):
-            file_ids -= set(directory.sudo().mapped("file_ids").ids)
-        return len(file_ids) if count else list(file_ids)
+            result -= set(directory.sudo().mapped("file_ids").ids)
+        return len(result) if count else list(result)
```

Comment:
I don't think this is going to work... `result` is a list (as returned from the original Odoo function https://github.com/odoo/odoo/blob/13.0/odoo/models.py) and you are trying to subtract a `set`, which is incompatible:
```
>>> [1, 2, 3] - {1}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'list' and 'set'
```
Maybe the subtracted item also needs to be a list, but I wonder if the conversion to `set` (that avoids duplicates) is necessary for this logic...

Reply:
IMO `_search()` function is not necessary now because only returned records that user can read. 
What do you think @etobella

Reply:
Well, that would be even better :sweat_smile: If it is possible, that option LGTM

Reply:
> IMO `_search()` function is not necessary now because only returned records that user can read.

That will only be true after #107, so we could consider adding the patch there, or keep this one as draft until then.